### DISCUSSION TITLE
Tiling not working. Am I doing something wrong?

### DISCUSSION THREAD

**Discussion by WhiteZeroX** [pkuliyi2015/multidiffusion-upscaler-for-automatic1111#186] [Q&A] [answered]
Trying out this add-on for the first time. I've used the Ultimate SD Upscale a lot and it works well, so I have some idea what I'm doing. However, when I run Tiled Diffusion, watching the preview window, it looks like it's not doing any tiling at all, it's only diffusing the whole image. Enable Tiled Diffusion is checked and the Noise Inversion step works fine from what I can see. But for some reason, no actual tiling is happening. I'm either doing something wrong or something is broken. If I uncheck "Keep input image size" and set img2img Resize By to something high like 3 or 4, I get an Out of Memory error, again because I think it's diffusing the entire image all at once instead of tiling it. 

This is with a freshly installed Tiled Diffusion extension in the latest A1111 git-pull (the Master branch, not the Dev branch or anything else). 

![Screenshot 2023-05-10 101609](https://github.com/pkuliyi2015/multidiffusion-upscaler-for-automatic1111/assets/16072090/210b0d3e-7a4e-4c2c-81ca-b2fc866906a5)
![Screenshot 2023-05-10 101628](https://github.com/pkuliyi2015/multidiffusion-upscaler-for-automatic1111/assets/16072090/c5d27cd9-3af9-4d0f-8dc1-3c36d437c0a9)

**Comment by JVirtual1** [comment]
I have the same issue and i won't even run on my rtx 4090 with "Keep input image size" enabled

**Reply by WhiteZeroX** [reply]
Maybe an update to the extension or A1111 broke something. Maybe I'll just try again later.

**Comment by Kahsolt** [comment]
if possible, please put a screenshot that can reveal which line crashed
and , "Keep input image size" is some what buggy, but seems like a webui bug, still no work around found :(

**Reply by WhiteZeroX** [reply]
Here are my console logs when it crashes. This is in img2img. 512x512 input image with 3x upscale. Tiled Diffusion on with "Keep input image size" turned OFF, Tiled VAE off, ControlNet Tile on, basically all of the same settings from my above screenshots, except for Keep Input Image Size. No other scripts turned on. 


```
[Tiled Diffusion] ControlNet found, support is enabled.████████████████████████████████| 20/20 [00:02<00:00,  6.29it/s]
Loading model: control_v11f1e_sd15_tile [a371b31b]
Loaded state_dict from [stable-diffusion-webui\extensions\sd-webui-controlnet\models\control_v11f1e_sd15_tile.pth]
Loading config: stable-diffusion-webui\extensions\sd-webui-controlnet\models\control_v11f1e_sd15_tile.yaml
ControlNet model control_v11f1e_sd15_tile [a371b31b] loaded.
Loading preprocessor: tile_resample
Pixel Perfect Mode Enabled.
resize_mode = ResizeMode.RESIZE
raw_H = 512
raw_W = 512
target_H = 1536
target_W = 1536
estimation = 1536.0
preprocessor resolution = 1536
MixtureOfDiffusers Sampling:   0%|                                                              | 0/60 [00:00<?, ?it/s]Mixture of Diffusers hooked into 'Euler' sampler, Tile size: 96x96, Tile batches: 3, Batch size: 4. (ext: ContrlNet)
Error completing request
Arguments: ('task(bns31zat080xkae)', 0, '', 'blurry', [], <PIL.Image.Image image mode=RGBA size=512x512 at 0x22C7F952230>, None, None, None, None, None, None, 20, 1, 4, 0, 1, False, False, 1, 1, 4.5, 1.5, 0.75, -1.0, -1.0, 0, 0, 0, False, 1, 512, 512, 3, 0, 0, 32, 0, '', '', '', [], 0, True, 'Mixture of Diffusers', False, False, 1024, 1024, 96, 96, 32, 4, 'None', 2, False, 20, 1, 0, 64, False, False, False, False, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 2048, 128, True, True, True, False, <controlnet.py.UiControlNetUnit object at 0x0000022C7FAF22C0>, <controlnet.py.UiControlNetUnit object at 0x0000022C7FABC6D0>, <controlnet.py.UiControlNetUnit object at 0x0000022C7F950190>, '<ul>\n<li><code>CFG Scale</code> should be 2 or lower.</li>\n</ul>\n', True, True, '', '', True, 50, True, 1, 0, False, 4, 0.5, 'Linear', 'None', '<p style="margin-bottom:0.75em">Recommended settings: Sampling Steps: 80-100, Sampler: Euler a, Denoising strength: 0.8</p>', 128, 8, ['left', 'right', 'up', 'down'], 1, 0.05, 128, 4, 0, ['left', 'right', 'up', 'down'], False, False, 'positive', 'comma', 0, False, False, '', '<p style="margin-bottom:0.75em">Will upscale the image by the selected scale factor; use width and height sliders to set tile size</p>', 64, 0, 2, 1, '', [], 0, '', [], 0, '', [], True, False, False, False, 0, None, False, None, False, None, False, 50, '<p style="margin-bottom:0.75em">Will upscale the image depending on the selected target size type</p>', 512, 0, 8, 32, 64, 0.35, 32, 0, True, 0, False, 8, 0, 0, 2048, 2048, 2) {}
Traceback (most recent call last):
  File "stable-diffusion-webui\modules\call_queue.py", line 57, in f
    res = list(func(*args, **kwargs))
  File "stable-diffusion-webui\modules\call_queue.py", line 37, in f
    res = func(*args, **kwargs)
  File "stable-diffusion-webui\modules\img2img.py", line 181, in img2img
    processed = process_images(p)
  File "stable-diffusion-webui\modules\processing.py", line 515, in process_images
    res = process_images_inner(p)
  File "stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\batch_hijack.py", line 42, in processing_process_images_hijack
    return getattr(processing, '__controlnet_original_process_images_inner')(p, *args, **kwargs)
  File "stable-diffusion-webui\modules\processing.py", line 604, in process_images_inner
    p.init(p.all_prompts, p.all_seeds, p.all_subseeds)
  File "stable-diffusion-webui\modules\processing.py", line 1084, in init
    self.init_latent = self.sd_model.get_first_stage_encoding(self.sd_model.encode_first_stage(image))
  File "stable-diffusion-webui\modules\sd_hijack_utils.py", line 17, in <lambda>
    setattr(resolved_obj, func_path[-1], lambda *args, **kwargs: self(*args, **kwargs))
  File "stable-diffusion-webui\modules\sd_hijack_utils.py", line 28, in __call__
    return self.__orig_func(*args, **kwargs)
  File "stable-diffusion-webui\venv\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "stable-diffusion-webui\repositories\stable-diffusion-stability-ai\ldm\models\diffusion\ddpm.py", line 830, in encode_first_stage
    return self.first_stage_model.encode(x)
  File "stable-diffusion-webui\repositories\stable-diffusion-stability-ai\ldm\models\autoencoder.py", line 83, in encode
    h = self.encoder(x)
  File "stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "stable-diffusion-webui\repositories\stable-diffusion-stability-ai\ldm\modules\diffusionmodules\model.py", line 536, in forward
    h = self.mid.attn_1(h)
  File "stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "stable-diffusion-webui\modules\sd_hijack_optimizations.py", line 490, in sdp_attnblock_forward
    out = torch.nn.functional.scaled_dot_product_attention(q, k, v, dropout_p=0.0, is_causal=False)
torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 2.53 GiB (GPU 0; 12.00 GiB total capacity; 8.62 GiB already allocated; 222.24 MiB free; 9.50 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF


```

**Reply by WhiteZeroX** [reply]
If I run it at 2x upscale I can complete the img2img fine, but I can do that without Tiled Diffusion. Because, again, I think Tiled Diffusion isn't actually splitting the image into tiles and just doing the whole thing. 


```
[Mem] rss: 6.980 GB, vms: 12.142 GB
[VRAM] free: 7.502 GB, total: 12.000 GB
[Tiled Diffusion] ControlNet found, support is enabled.
Loading model from cache: control_v11f1e_sd15_tile [a371b31b]
Loading preprocessor: tile_resample
Pixel Perfect Mode Enabled.
resize_mode = ResizeMode.RESIZE
raw_H = 512
raw_W = 512
target_H = 1024
target_W = 1024
estimation = 1024.0
preprocessor resolution = 1024
warn: noise inversion only supports the Euler sampler, switch to it sliently...
MixtureOfDiffusers Sampling:   0%|                                                              | 0/20 [00:00<?, ?it/s]Mixture of Diffusers hooked into 'Euler' sampler, Tile size: 96x96, Tile batches: 1, Batch size: 4. (ext: NoiseInv, ContrlNet)
Noise Inversion: 100%|█████████████████████████████████████████████████████████████████| 20/20 [00:10<00:00,  1.84it/s]
100%|██████████████████████████████████████████████████████████████████████████████████| 16/16 [00:13<00:00,  1.19it/s]
Total progress:  50%|█████████████████████████████████                                 | 16/32 [00:13<00:13,  1.19it/s]
Total progress:  50%|█████████████████████████████████                                 | 16/32 [00:13<00:13,  1.19it/s] 
```

**Reply by PotatoBananaApple** [reply]
Try using higher tile size. I have noticed, sometimes i run out of memory randomly without any reason with very specific settings.

**Reply by WhiteZeroX** [reply]
> Try using higher tile size. I have noticed, sometimes i run out of memory randomly without any reason with very specific settings.

I have, no difference. This is because, again, I'm pretty sure no actual Tiling is occurring, it's just doing the whole image.

**Reply by WhiteZeroX** [reply]
Am I dumb and you HAVE to use Tiled Diffusion with something like Ultimate SD Upscale which will actually do the tiling? I'm assuming this extension is suppose to do tiling on it's own??

**Reply by pkuliyi2015** [reply]
From your logs, it is obvious that you didn’t enable the tiled VAE. The two functions need to work together to prevent OOM.

**Reply by WhiteZeroX** [reply]
> From your logs, it is obvious that you didn’t enable the tiled VAE. The two functions need to work together to prevent OOM.

Here ya go, logs with Tiled VAE enabled. 

```
[Mem] rss: 9.438 GB, vms: 15.392 GB
[VRAM] free: 6.617 GB, total: 12.000 GB
[Tiled Diffusion] ControlNet found, support is enabled.
Loading model from cache: control_v11f1e_sd15_tile [a371b31b]
Loading preprocessor: tile_resample
Pixel Perfect Mode Enabled.
resize_mode = ResizeMode.RESIZE
raw_H = 512
raw_W = 640
target_H = 1536
target_W = 1920
estimation = 1536.0
preprocessor resolution = 1536
MixtureOfDiffusers Sampling:   0%|                                                              | 0/42 [00:00<?, ?it/s]Mixture of Diffusers hooked into 'Euler' sampler, Tile size: 96x96, Tile batches: 3, Batch size: 4. (ext: ContrlNet)
[Tiled VAE]: the input size is tiny and unnecessary to tile.
Error completing request
Arguments: ('task(o0jabm4ev29e01y)', 0, ' <lora:COOLKIDS_MERGE_V2.5:0.5>', 'nsfw, blurry, poorly drawn, ugly, deformed, fused, hazy,   ng_deepnegative_v1_75t', [], <PIL.Image.Image image mode=RGBA size=640x512 at 0x1DF80B532E0>, None, None, None, None, None, None, 20, 1, 4, 0, 1, False, False, 1, 1, 6, 1.5, 0.6, -1.0, -1.0, 0, 0, 0, False, 1, 512, 512, 3, 0, 0, 32, 0, '', '', '', [], 0, True, 'Mixture of Diffusers', False, False, 1024, 1024, 96, 96, 32, 4, 'None', 2, False, 20, 1, 0, 64, False, False, False, False, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, True, 2048, 128, True, True, True, False, <controlnet.py.UiControlNetUnit object at 0x000001E0795B0C70>, <controlnet.py.UiControlNetUnit object at 0x000001DF80CBF7C0>, <controlnet.py.UiControlNetUnit object at 0x000001DF80CBC970>, '<ul>\n<li><code>CFG Scale</code> should be 2 or lower.</li>\n</ul>\n', True, True, '', '', True, 50, True, 1, 0, False, 4, 0.5, 'Linear', 'None', '<p style="margin-bottom:0.75em">Recommended settings: Sampling Steps: 80-100, Sampler: Euler a, Denoising strength: 0.8</p>', 128, 8, ['left', 'right', 'up', 'down'], 1, 0.05, 128, 4, 0, ['left', 'right', 'up', 'down'], False, False, 'positive', 'comma', 0, False, False, '', '<p style="margin-bottom:0.75em">Will upscale the image by the selected scale factor; use width and height sliders to set tile size</p>', 64, 0, 2, 1, '', [], 0, '', [], 0, '', [], True, False, False, False, 0, None, False, None, False, None, False, 50, '<p style="margin-bottom:0.75em">Will upscale the image depending on the selected target size type</p>', 680, 0, 8, 32, 64, 0.35, 32, 0, True, 0, False, 8, 0, 0, 2048, 2048, 2) {}
Traceback (most recent call last):
  File "C:\Games\Misc\stablediffusion\stable-diffusion-webui\modules\call_queue.py", line 57, in f
    res = list(func(*args, **kwargs))
  File "C:\Games\Misc\stablediffusion\stable-diffusion-webui\modules\call_queue.py", line 37, in f
    res = func(*args, **kwargs)
  File "C:\Games\Misc\stablediffusion\stable-diffusion-webui\modules\img2img.py", line 181, in img2img
    processed = process_images(p)
  File "C:\Games\Misc\stablediffusion\stable-diffusion-webui\modules\processing.py", line 515, in process_images
    res = process_images_inner(p)
  File "C:\Games\Misc\stablediffusion\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\batch_hijack.py", line 42, in processing_process_images_hijack
    return getattr(processing, '__controlnet_original_process_images_inner')(p, *args, **kwargs)
  File "C:\Games\Misc\stablediffusion\stable-diffusion-webui\modules\processing.py", line 604, in process_images_inner
    p.init(p.all_prompts, p.all_seeds, p.all_subseeds)
  File "C:\Games\Misc\stablediffusion\stable-diffusion-webui\modules\processing.py", line 1084, in init
    self.init_latent = self.sd_model.get_first_stage_encoding(self.sd_model.encode_first_stage(image))
  File "C:\Games\Misc\stablediffusion\stable-diffusion-webui\modules\sd_hijack_utils.py", line 17, in <lambda>
    setattr(resolved_obj, func_path[-1], lambda *args, **kwargs: self(*args, **kwargs))
  File "C:\Games\Misc\stablediffusion\stable-diffusion-webui\modules\sd_hijack_utils.py", line 28, in __call__
    return self.__orig_func(*args, **kwargs)
  File "C:\Games\Misc\stablediffusion\stable-diffusion-webui\venv\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "C:\Games\Misc\stablediffusion\stable-diffusion-webui\repositories\stable-diffusion-stability-ai\ldm\models\diffusion\ddpm.py", line 830, in encode_first_stage
    return self.first_stage_model.encode(x)
  File "C:\Games\Misc\stablediffusion\stable-diffusion-webui\repositories\stable-diffusion-stability-ai\ldm\models\autoencoder.py", line 83, in encode
    h = self.encoder(x)
  File "C:\Games\Misc\stablediffusion\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "C:\Games\Misc\stablediffusion\stable-diffusion-webui\extensions\multidiffusion-upscaler-for-automatic1111\scripts\vae_optimize.py", line 379, in __call__
    return self.net.original_forward(x)
  File "C:\Games\Misc\stablediffusion\stable-diffusion-webui\repositories\stable-diffusion-stability-ai\ldm\modules\diffusionmodules\model.py", line 536, in forward
    h = self.mid.attn_1(h)
  File "C:\Games\Misc\stablediffusion\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "C:\Games\Misc\stablediffusion\stable-diffusion-webui\modules\sd_hijack_optimizations.py", line 490, in sdp_attnblock_forward
    out = torch.nn.functional.scaled_dot_product_attention(q, k, v, dropout_p=0.0, is_causal=False)
torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 3.96 GiB (GPU 0; 12.00 GiB total capacity; 7.05 GiB already allocated; 1.53 GiB free; 8.12 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF

```

**Reply by PotatoBananaApple** [reply]
Checked the image again and noticed you have no upscaler selected in tiled diffusion. And the log does not mention the upscaling with tiled diffusion (unless i am blind). Is there reason you are using `Resize to/by` in img2img?  Managed to get the same error fiddling with resize to/by settings. Check if it works if you just upscale with the tiled diffusion.

Also, do you have xformers enabled? It reduces VRAM usage.

**Reply by WhiteZeroX** [reply]
> Checked the image again and noticed you have no upscaler selected in tiled diffusion. And the log does not mention the upscaling with tiled diffusion (unless i am blind). Is there reason you are using `Resize to/by` in img2img? Managed to get the same error fiddling with resize to/by settings. Check if it works if you just upscale with the tiled diffusion.
> 
> Also, do you have xformers enabled? It reduces VRAM usage.

Instead of xformers I was using --opt-sdp-attention --opt-split-attention, which seemed to be similar. If I switch back to xformers it seems like I can barely get a 3x upscale to work, but it fails at 4x. 

Can you clarify if the Preview window during diffusing should show it working each tile individually? Seeing each tile diffuse is something I can see when Ultimate SD Upscale is working, but when I used TIled Diffusion it shows the entire image, which is what makes me think it's not actually tiling anything. See this ref image:

![Diffusing](https://github.com/pkuliyi2015/multidiffusion-upscaler-for-automatic1111/assets/16072090/d4c9791f-76e9-4d54-9a0e-980b443ec123)


If I leave Resize By to 1 and instead use an upscaler like ESRGAN, I get the same Out of Memory error at the same point. Meanwhile with Ultimate SD Upscale, I can do 5x or more, because it's actually splitting the job into tiles and it maxes out at like 6GB of VRAM.

**Reply by PotatoBananaApple** [reply]
It is hard to see in the preview window, controlnet tiling resample makes it even harder and if you use mixture of diffusers it is pretty impossible. You can interupt the generate after 1 preview and you should see the tiles in the image it gives. As far as i can understand, it seems like it generates the tiles and combines the tiles before updating the preview.
![image](https://github.com/pkuliyi2015/multidiffusion-upscaler-for-automatic1111/assets/130109414/b519533e-299a-41f2-a4d5-baa799d962b3)

Doing some testing and i can run 7000x4000 image with 12gb of VRAM using 8.1GB at start. I noticed spike going up in the VRAM usage at one point. I think it was when it generated the preview picture first time. It went up 0.8gb with Approx NN preview. If you are using full preview and get spike with that, it could be cause of the memory error.


Yeah definitely could be problem with the preview. If Ultimate SD Upscale divides them while doing preview, that could be the difference here.
Here tested with full preview, on pretty low upscale. The spike is real 🤣
![image](https://github.com/pkuliyi2015/multidiffusion-upscaler-for-automatic1111/assets/130109414/12789994-edd5-4afb-9a3c-450a675e06bc)

Setting VAE tile size to really low values makes the spike lower
![image](https://github.com/pkuliyi2015/multidiffusion-upscaler-for-automatic1111/assets/130109414/efd948d5-447c-4bd7-bed4-2d15bb83114a)

**Reply by WhiteZeroX** [reply]
Seems you're right about interrupting the diffuser to see the tiles. If I interrupt or set Steps to 1, I can see the grid of tiles. So I guess it is working in tiles after all. I do also have Approx NN preview turned on, never use Full. I also tried with Previews off, but I'm still getting Out of Memory with anything greater than like 1950 x 1950 on my 12GB card. So something still seems wrong with my setup.

**Reply by PotatoBananaApple** [reply]
> Seems you're right about interrupting the diffuser to see the tiles. If I interrupt or set Steps to 1, I can see the grid of tiles. So I guess it is working in tiles after all. I do also have Approx NN preview turned on, never use Full. I also tried with Previews off, but I'm still getting Out of Memory with anything greater than like 1950 x 1950 on my 12GB card. So something still seems wrong with my setup.

Difference between your settings i see and my settings are:
I don't use noise inversion
Low VRAM disabled on controlnet,
Pixel perfect disabled, i did not see any difference between this on/off. 
Resize mode on controlnet: Crop and resize, does not really matter if your image size matches with the width and height of the generate
The only resize i do is with the upscale from tiled diffusion (Upscaler and Scale Factor)

**Reply by WhiteZeroX** [reply]
> > Seems you're right about interrupting the diffuser to see the tiles. If I interrupt or set Steps to 1, I can see the grid of tiles. So I guess it is working in tiles after all. I do also have Approx NN preview turned on, never use Full. I also tried with Previews off, but I'm still getting Out of Memory with anything greater than like 1950 x 1950 on my 12GB card. So something still seems wrong with my setup.
> 
> Difference between your settings i see and my settings are: I don't use noise inversion Low VRAM disabled on controlnet, Pixel perfect disabled, i did not see any difference between this on/off. Resize mode on controlnet: Crop and resize, does not really matter if your image size matches with the width and height of the generate The only resize i do is with the upscale from tiled diffusion (Upscaler and Scale Factor)

I gave it a try with these settings and have the same issue, unfortunately.

**Reply by pkuliyi2015** [reply, marked as answer]
Please reduce the tiled vae size. It is not successfully activated Because the vae tile size settings is too large.

**Reply by WhiteZeroX** [reply]
> Please reduce the tiled vae size. It is not successfully activated Because the vae tile size settings is too large.

Ah! That seems to have been my issue! I lowered the Tiled VAE resolutions to half their default and I can now upscale significantly higher. I guess I assumed the defaults were somewhat conservative to begin with, but now I know better.


```
[Tiled Diffusion] ControlNet found, support is enabled.
Loading model from cache: control_v11f1e_sd15_tile [a371b31b]
Loading preprocessor: tile_resample
Pixel Perfect Mode Enabled.
resize_mode = ResizeMode.RESIZE
raw_H = 2560
raw_W = 2560
target_H = 2560
target_W = 2560
estimation = 2560.0
preprocessor resolution = 2560
warn: noise inversion only supports the Euler sampler, switch to it sliently...
MultiDiffusion Sampling:   0%|                                                                 | 0/144 [00:00<?, ?it/s]MultiDiffusion hooked into 'Euler' sampler, Tile size: 96x96, Tile batches: 9, Batch size: 4. (ext: NoiseInv, ContrlNet)
MultiDiffusion Sampling:  83%|█████████████████████████████████████████████▊         | 120/144 [12:51<02:34,  6.43s/it]
[Tiled VAE]: input_size: torch.Size([1, 3, 2560, 2560]), tile_size: 1024, padding: 32
[Tiled VAE]: split to 3x3 = 9 tiles. Optimal tile size 832x832, original tile size 1024x1024
[Tiled VAE]: Fast mode enabled, estimating group norm parameters on 1024 x 1024 image
[Tiled VAE]: Executing Encoder Task Queue: 100%|████████████████████████████████████| 819/819 [00:01<00:00, 811.77it/s]
[Tiled VAE]: Done in 1.789s, max VRAM alloc 4635.967 MB█████████████████████████    | 729/819 [00:00<00:00, 810.88it/s]
MultiDiffusion Sampling: 100%|███████████████████████████████████████████████████████| 144/144 [01:29<00:00,  1.61it/s]
Noise Inversion: 100%|█████████████████████████████████████████████████████████████████| 30/30 [02:25<00:00,  4.85s/it]
100%|██████████████████████████████████████████████████████████████████████████████████| 16/16 [01:58<00:00,  7.43s/it]
[Tiled VAE]: input_size: torch.Size([1, 4, 320, 320]), tile_size: 128, padding: 11     | 16/32 [01:51<01:58,  7.41s/it]
[Tiled VAE]: split to 3x3 = 9 tiles. Optimal tile size 128x128, original tile size 128x128
[Tiled VAE]: Fast mode enabled, estimating group norm parameters on 128 x 128 image
[Tiled VAE]: Executing Decoder Task Queue: 100%|██████████████████████████████████| 1107/1107 [00:05<00:00, 206.52it/s]
[Tiled VAE]: Done in 6.364s, max VRAM alloc 7637.538 MB
Total progress:  50%|█████████████████████████████████                                 | 16/32 [01:59<01:59,  7.45s/it]
Total progress:  50%|█████████████████████████████████                                 | 16/32 [01:59<01:58,  7.41s/it] 
```

**Comment by Kahsolt** [comment]
This is due to webui scripts execution order, the controlnet stuff is processed before tiled-vae hooks on, so that tiled-vae failed to hijack the VAE encoding, leading to OOM.

Need a work around....
see also #190

**Reply by WhiteZeroX** [reply]
> This is due to webui scripts execution order, the controlnet stuff is processed before tiled-vae hooks on, so that tiled-vae failed to hijack the VAE encoding, leading to OOM.
> 
> Need a work around.... see also #190

Interestingly now that I lowered my Tiled VAE resolutions, it now seems to work fine.

**Comment by Rkkss** [comment]
btw can we use tiled diffusion to regional upscale since it's compatible with inpaint.
I always does **inpainted not masked/whole image** if I don't want certain part of image get upscaled and it indeed work

But when I does **inpaint masked/only masked**, console gives `ValueError: images do not match` error. Am I doing wrong or you must set scale factor to 1 when do **only masked**?
![Untitled](https://github.com/pkuliyi2015/multidiffusion-upscaler-for-automatic1111/assets/127614350/dc456113-608a-47bb-9fa2-c8401f7a4164)

**Reply by PotatoBananaApple** [reply]
Yes scale factor must be 1 and put upscaler: none. It affects whole image if set to something else.
